server/runtime/src/main/java/org/infinispan/server/Bootstrap.java

```diff
@@ -101,8 +104,17 @@ protected void handleArgumentCommand(String command, String parameter, Iterator<
          case "--port-offset":
             properties.setProperty(Server.INFINISPAN_PORT_OFFSET, parameter);
             break;
+         case "-P":
+            parameter = args.next();
+         case "--properties":
+            try(Reader r = Files.newBufferedReader(Paths.get(parameter))) {
+               properties.load(r);
```

Comment:
-1
this will overwrite everything defined with `-D<key>=<value>` (and can overwrite other arguments as well like `-p`, `-o`)

Reply:
Order matters.

Reply:
order doesn't matter. `-D` properties are always parser before the `Bootstrap.handleArgumentCommand()`

```
$ cat keys.properties 
infinispan.socket.binding.port-offset=100
$ /bin/server.sh -P keys.properties -Dinfinispan.socket.binding.port-offset=200
...
10:09:34,212 INFO  (main) [org.infinispan.SERVER] ISPN080004: Connector SINGLE_PORT (default) listening on 127.0.0.1:11322
```

Reply:
Right. I'll do a merge instead of a load.